src/clr-core/button/button.element.ts

```diff
@@ -34,162 +32,58 @@ export enum ClrLoadingState {
- * @cssprop --clr-btn-success-checked-color
- * @cssprop --clr-btn-success-outline-color
- * @cssprop --clr-btn-success-outline-bg-color
- * @cssprop --clr-btn-success-outline-border-color
- * @cssprop --clr-btn-success-outline-hover-bg-color
- * @cssprop --clr-btn-success-outline-hover-color
- * @cssprop --clr-btn-success-outline-box-shadow-color
- * @cssprop --clr-btn-success-outline-disabled-color
- * @cssprop --clr-btn-success-outline-disabled-bg-color
- * @cssprop --clr-btn-success-outline-disabled-border-color
- * @cssprop --clr-btn-success-outline-checked-bg-color
- * @cssprop --clr-btn-success-outline-checked-color
- * @cssprop --clr-btn-danger-color
- * @cssprop --clr-btn-danger-bg-color
- * @cssprop --clr-btn-danger-border-color
- * @cssprop --clr-btn-danger-hover-bg-color
- * @cssprop --clr-btn-danger-hover-color
- * @cssprop --clr-btn-danger-box-shadow-color
- * @cssprop --clr-btn-danger-disabled-color
- * @cssprop --clr-btn-danger-disabled-bg-color
- * @cssprop --clr-btn-danger-disabled-border-color
- * @cssprop --clr-btn-danger-checked-bg-color
- * @cssprop --clr-btn-danger-checked-color
- * @cssprop --clr-btn-danger-outline-color
- * @cssprop --clr-btn-danger-outline-bg-color
- * @cssprop --clr-btn-danger-outline-border-color
- * @cssprop --clr-btn-danger-outline-hover-bg-color
- * @cssprop --clr-btn-danger-outline-hover-color
- * @cssprop --clr-btn-danger-outline-box-shadow-color
- * @cssprop --clr-btn-danger-outline-disabled-color
- * @cssprop --clr-btn-danger-outline-disabled-bg-color
- * @cssprop --clr-btn-danger-outline-disabled-border-color
- * @cssprop --clr-btn-danger-outline-checked-bg-color
- * @cssprop --clr-btn-danger-outline-checked-color
- * @cssprop --clr-btn-link-color
- * @cssprop --clr-btn-link-bg-color
- * @cssprop --clr-btn-link-border-color
- * @cssprop --clr-btn-link-hover-bg-color
- * @cssprop --clr-btn-link-hover-color
- * @cssprop --clr-btn-link-disabled-color
- * @cssprop --clr-btn-link-disabled-bg-color
- * @cssprop --clr-btn-link-disabled-border-color
- * @cssprop --clr-btn-link-checked-bg-color
- * @cssprop --clr-btn-link-checked-color
- * @cssprop --clr-btn-inverse-color
- * @cssprop --clr-btn-inverse-border-color
- * @cssprop --clr-btn-inverse-bg-color
- * @cssprop --clr-btn-inverse-hover-bg-color
- * @cssprop --clr-btn-inverse-hover-color
- * @cssprop --clr-btn-inverse-box-shadow-color
- * @cssprop --clr-btn-inverse-disabled-color
- * @cssprop --clr-btn-inverse-disabled-bg-color
- * @cssprop --clr-btn-inverse-disabled-border-color
- * @cssprop --clr-btn-inverse-checked-bg-color
- * @cssprop --clr-btn-inverse-checked-color
- * @cssprop --clr-btn-icon-disabled-color
- * @cssprop --clr-btn-group-focus-outline
+ * @cssprop --box-shadow-color
+ * @cssprop --border-radius
+ * @cssprop --border-width
+ * @cssprop --font-size
+ * @cssprop --font-weight
+ * @cssprop --border-color
+ * @cssprop --color
+ * @cssprop --background
+ * @cssprop --line-height
+ * @cssprop --letter-spacing
```

Comment:
I applied the technique we started to use to modify different states with CSS properties in badges/tags to the button component. This made the API much surface much smaller. https://github.com/vmware/clarity/issues/4062